src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -1432,6 +1465,22 @@ def _handle_addons_args(cmd, addons_str, subscription_id, resource_group_name, a
     if 'azure-policy' in addons:
         addon_profiles['azurepolicy'] = ManagedClusterAddonProfile(enabled=True)
         addons.remove('azure-policy')
+    if 'ingress-appgw' in addons:
+        addon_profile = ManagedClusterAddonProfile(enabled=True, config={})
+        if appgw_name is not None:
+            addon_profile.config["applicationGatewayName"] = appgw_name
+        if appgw_subnet_prefix is not None:
+            addon_profile.config["subnetPrefix"] = appgw_subnet_prefix
+        if appgw_id is not None:
+            addon_profile.config["applicationGatewayId"] = appgw_id
+        if appgw_subnet_id is not None:
+            addon_profile.config["subnetId"] = appgw_subnet_id
+        if appgw_shared:
+            addon_profile.config["shared"] = "true"
+        if appgw_watch_namespace is not None:
+            addon_profile.config["watchNamespace"] = appgw_watch_namespace
```

Comment:
Advice on code style: Maybe you can assign these keys to constant variables. It's error-prone if duplicate string literals are all over the code.